### PR TITLE
Publish split-release artifacts more granularly

### PR DESCRIPTION
**Pull request by cocreature**
This publishes damlc & the Daml libraries as standalone artifacts so
Canton can avoid downloading the whole SDK tarball (hopefully).

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ci/copy-unix-release-artifacts.sh [hunk 1]
```diff
@@ -9,15 +9,19 @@ OUTPUT_DIR=$3
 
 mkdir -p $OUTPUT_DIR/github
 mkdir -p $OUTPUT_DIR/artifactory
+# Artifacts that we only use in the split-release process
+mkdir -p $OUTPUT_DIR/split-release
 
 
 TARBALL=daml-sdk-$RELEASE_TAG-$NAME.tar.gz
 EE_TARBALL=daml-sdk-$RELEASE_TAG-$NAME-ee.tar.gz
+bazel build //release:sdk-release-tarball-ce //release:sdk-release-tarball-ee
```
Comment (garyverhaegen-da) [inline ci/copy-unix-release-artifacts.sh:18]: Why are those suddenly needed? Local testing?

File: ci/copy-unix-release-artifacts.sh [hunk 2]
```diff
@@ -83,5 +87,14 @@ if [[ "$NAME" == "linux" ]]; then
     cp bazel-bin/runtime-components/non-repudiation-client/libnon-repudiation-client-src.jar $OUTPUT_DIR/artifactory/$NON_REPUDIATION_CLIENT_SRC
     cp bazel-bin/runtime-components/non-repudiation-client/non-repudiation-client_javadoc.jar $OUTPUT_DIR/artifactory/$NON_REPUDIATION_CLIENT_DOC
 
-fi
+    bazel build //compiler/damlc:damlc-dist
+    cp bazel-bin/compiler/damlc/damlc-dist.tar.gz $OUTPUT_DIR/split-release/damlc-$RELEASE_TAG.tar.gz
+
+    mkdir -p $OUTPUT_DIR/split-release/daml-libs/daml-script
+    bazel build //daml-script/daml:daml-script-dars
+    cp bazel-bin/daml-script/daml/*.dar $OUTPUT_DIR/split-release/daml-libs/daml-script/
```
Comment (garyverhaegen-da) [inline ci/copy-unix-release-artifacts.sh:95]: Will those include version numbers too? (Should they?)

**Review by garyverhaegen-da — APPROVED** [review]
Thanks!

File: ci/copy-unix-release-artifacts.sh (hunk 1 above)
Comment (cocreature) [reply ci/copy-unix-release-artifacts.sh:18]: yeah just local testing. seemed easier to build everything than to rely on stuff having been built before by a CI job.

File: ci/copy-unix-release-artifacts.sh (hunk 2 above)
Comment (cocreature) [reply ci/copy-unix-release-artifacts.sh:95]: I considered it but in the end it seemed nice to keep the name they have in the SDK tarball. That way you can in theory make `daml-script` as a dependency work by putting it in the right directory and set the `DAML_SDK` env var even if you just have those dars and not the rest of the SDK. We do include the version in the url.

Comment (garyverhaegen-da) [reply ci/copy-unix-release-artifacts.sh:95]: That makes sense, thanks for explaining.

**Pull request merged**